crates/conformance/src/lib.rs

```diff
@@ -0,0 +1,936 @@
+//! Spin Conformance Tests
+//!
+//! This crate provides a suite of tests to determine which features of Spin a guest module is capable of
+//! exercising.  The suite may be run via the [`test`] function, which accepts a [`wasmtime::Module`] and returns a
+//! [`Report`] which details which features the module does or does not support.
```

Comment:
Yeah, I've struggled with how to phrase this.  It basically has to do two things:

- Verify trigger functions are exported from the guest and do something sane when called.
- Verify guest imports host-implemented functions and calls them with the parameters specified by the test

The latter of course can only be tested by first calling into the guest so it can call back to the host.  As of my latest commit this can either be done via WASI's `_start` or by spin-http.wit's `handle-http-request`.

I'm open to suggestions on rephrasing this.

Reply:
@itowlson I went ahead and reworded this.  Please let me know what you think.